pyquil/api/_qpu.py

```diff
@@ -142,6 +166,10 @@ def _build_patch_values(self) -> dict:
                 continue
             patch_table[name] = [0] * spec.length
 
+        mref_name = self._build_arithmetic_patch_values()
+        if mref_name:
```

Comment:
I really don't like these "implicit boolean" conditions. Can we change to `if mref_name is not None` or `if mref_name != ""`. also see comment below on multiple return types for build_arithmetic_patch_values